### PR TITLE
feat: add new Icons package for custom made icons

### PR DESCRIPTION
**Pull request by MildTomato**
## What kind of change does this PR introduce?

New icons package for custom icons for Supabase apps.

- adds new package, `./packages/icons`
- build scripts are in `./packages/icons/internal/build-icons`
  - this takes raw SVG files and then converts them into react friendly components
  - Used the same build scripts as lucide

### example usecase  

```js
import { ReplaceCode, InsertCode } from 'icons'

function app() {
  return (
    <>
        <ReplaceCode className="text-light" strokeWidth={1} size={16} aria-expanded={true} />
        <InsertCode className="text-light" strokeWidth={1} size={16} />
        <Database className="text-light" strokeWidth={1} size={16} />
        <Auth className="text-light" strokeWidth={1} size={16} />
        <EdgeFunctions className="text-light" strokeWidth={1} size={16} />
        <Integrations className="text-light" strokeWidth={1} size={16} />
        <Realtime className="text-light" strokeWidth={1} size={16} />
        <Reports className="text-light" strokeWidth={1} size={16} />
        <RESTApi className="text-light" strokeWidth={1} size={16} />
        <Settings className="text-light" strokeWidth={1} size={16} />
        <TableEditor className="text-light" strokeWidth={1} size={16} />
        <SqlEditor className="text-light" strokeWidth={1} size={16} />
        <Storage className="text-light" strokeWidth={1} size={16} />
        <Logs className="text-light" strokeWidth={1} size={16} />
        <User className="text-light" strokeWidth={1} size={16} />
        <ApiDocs className="text-light" strokeWidth={1} size={16} />
    </>
)}
```

Screenshot 2024-02-23 at 3 50 13 PM

## what next

- follow up this PR with changes to use these icons in menus for Studio and Docs site

## Additional context

- it's possible we will deprecate the icons in ./ui package as they are becoming a bit redundant

### CODE REVIEW

File: packages/icons/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      <ReplaceCode className="text-light" strokeWidth={1} size={16} />
+      <InsertCode className="text-light" strokeWidth={1} size={16} />
+    </>
+  )
+}
+```
+
+## adding new icons
+
+Add new icons into ./src/raw-icons
```
Comment (fsansalvadore) [inline packages/icons/README.md:23]: @MildTomato these icons should always be exported 24x24px and have an icon that's around 18x18px(ish) for consistency, right? Might be worth mentioning in the README.md

Comment (MildTomato) [reply packages/icons/README.md:23]: adding

Comment (MildTomato) [reply packages/icons/README.md:23]: done

**Review by ivasilov — APPROVED** [review]

**Comment by MildTomato** [comment]
@fsansalvadore you ok to unblock this review?

**Review by fsansalvadore — APPROVED** [review]
Nice job 🔥

**Pull request merged**